src/matter/interaction/InteractionServer.ts

```diff
@@ -168,6 +171,74 @@ export class InteractionServer implements ProtocolHandler<MatterDevice> {
         };
     }
 
+    handleWriteRequest(exchange: MessageExchange<MatterDevice>, {suppressResponse, writeRequests }: WriteRequest): WriteResponse | undefined {
+        logger.debug(`Received write request from ${exchange.channel.getName()} for ${writeRequests.length} attributes, suppressResponse=${suppressResponse}`);
+
+        // TODO consider TimedRequest constrains
+
+        const writeResponses = [];
+        for (const request of writeRequests) {
```

Comment:
Do you have an info how to get info if an attribute is writable or the min/maxlength?

Reply:
set() calls the validator() method internally so you don't have to about min / max or other data constraints

Maybe let's do the same for writable:
* add a writable flag the AttributeServer which is populated at creation with the value from the attribute definition
* check / throw error in the set() method is called on a non-writable attribute (but not on setLocal since the value can be changed by the local code)

If we want to return the correct error Status, we can add an enum in the errors thrown indicating the error reason and convert it into the correct error Status in the catch in the code above.